Sources/SwiftDocCTestUtilities/XCTestCase+TemporaryDirectory.swift

```diff
@@ -0,0 +1,55 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2021 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+import XCTest
+
+// These helpers methods exist to put temp files for different test executions in different locations when running in Swift CI.
+
+public extension XCTestCase {
+    
+    @available(*, deprecated, message: "Use `createTemporaryDirectory` instead in unit tests to avoid referencing a shared location in Swift CI.")
+    func NSTemporaryDirectory() -> String {
+        return Foundation.NSTemporaryDirectory()
+    }
+    
+    /// Creates a new temporary directory and returns the URL of that directory.
+    ///
+    /// After the current test method has returned the temporary directory is automatically removed.
+    ///
+    /// - Parameters:
+    ///   - pathComponents: Additional path components to add to the temporary URL.
+    ///   - fileManager: The file manager that will create the directory.
+    /// - Returns: The URL of the newly created directory.
+    func createTemporaryDirectory(pathComponents: String..., fileManager: FileManager = .default) throws -> URL {
```

Comment:
I think all of our current uses of this just specify a single path component because they need a specifically named temporary directory. Maybe this would be easier to understand at the callsite with something like-

```suggestion
    func createTemporaryDirectory(named directoryName: String? = nil, fileManager: FileManager = .default) throws -> URL {
```